backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/notification/impl/UserNotificationsConfigRepository.java

```diff
@@ -54,32 +58,78 @@
 
 public class UserNotificationsConfigRepository implements IUserNotificationsConfigRepository
 {
-	private final CCache<UserId, UserNotificationsConfig> userNotificationsConfigsByUserId = CCache.<UserId, UserNotificationsConfig> builder()
+	private final CCache<UserId, UserNotificationsConfig> userNotificationsConfigsByUserId = CCache.<UserId, UserNotificationsConfig>builder()
 			.tableName(I_AD_User_NotificationGroup.Table_Name)
 			.initialCapacity(100)
 			.cacheMapType(CacheMapType.LRU)
 			.additionalTableNameToResetFor(org.compiere.model.I_AD_User.Table_Name)
 			.additionalTableNameToResetFor(I_AD_NotificationGroup.Table_Name)
 			.build();
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final INotificationGroupNameRepository notificationGroupNamesRepo = Services.get(INotificationGroupNameRepository.class);
+	private final IUserDAO userDAO = Services.get(IUserDAO.class);
+	private final IBPartnerOrgBL bPartnerOrgBL = Services.get(IBPartnerOrgBL.class);
```

Comment:
looking at the amount of services introduced here in this Repository, IMHO that's a sign that those shall go to UserNotificationsConfigService or similar.

DAO/Repositories cannot have calls to other services/DAO/repositories.
in DDD, Repositories are pretty dumb.
Their only responsibility is to persist and retrieve/search aggregates.
They don't ask questions, they don't take decisions.

Reply:
dumbified repo and introduced UserNotificationsConfigService.